src/main/java/mchorse/metamorph/api/MorphSettings.java

```diff
@@ -20,58 +20,81 @@
  * 
  * An instance of this class is responsible for storing information about 
  * morph's configurable settings.
+ * 
+ * For well-defined defaults, use {@link #DEFAULT_MORPHED} or {@link #DEFAULT}.
  */
 public class MorphSettings
 {
     /**
-     * Empty morph settings which doesn't have any attributes 
+     * "Safe" settings, equivalent to not being morphed
      */
     public static final MorphSettings DEFAULT = new MorphSettings();
+    
+    static
+    {
+        DEFAULT.hostile = false;
+        DEFAULT.hands = true;
+        DEFAULT.updates = false;
+    }
+
+    /**
+     * Default settings to fall back on for most morphs.
+     */
+    public static final MorphSettings DEFAULT_MORPHED = new MorphSettings();
 
     /**
      * Abilities that are going to be applied on a morph 
      */
     public List<IAbility> abilities = new ArrayList<IAbility>();
+    public boolean hasAbilities = true;
 
     /**
      * Attack that is going to be used on a morph
      */
-    public IAttackAbility attack;
+    public IAttackAbility attack = null;
+    public boolean hasAttack = true;
 
     /**
      * Action that is going to be used on a morph
      */
-    public IAction action;
+    public IAction action = null;
+    public boolean hasAction = true;
 
     /**
      * Health units which are going to be applied
      */
     public int health = 20;
+    public boolean hasHealth = true;
 
     /**
      * Speed which is going to be applied 
      */
     public float speed = 0.1F;
+    public boolean hasSpeed = true;
 
     /**
      * Hostile flag which is going to be applied
      */
-    public boolean hostile;
+    public boolean hostile = true;
+    public boolean hasHostile = true;
 
     /**
      * Does client tries render hands for this morph 
      */
-    public boolean hands;
+    public boolean hands = false;
+    public boolean hasHands = true;
```

Comment:
I believe there is a built-in Java type called `Optional` which is suitable for this.

This will require updates in any add-ons which depend on these settings, though, which is why I didn't do it in Changeling, as it would prevent add-ons made for Metamorph to work with Changeling and vice-versa.

It is your call, however, with regard to compatibility, as you are the maintainer.

If we're willing to break compatibility, then we might as well make the settings field in `AbstractMorph` protected, to encourage use of `getSettings` instead, which is more robust.

Reply:
Let's continue writing on discord so we don't end up with endless messages.